app/code/Magento/Customer/Api/AccountManagementInterface.php

```diff
@@ -170,9 +171,24 @@ public function resetPassword($email, $resetToken, $newPassword);
      * @throws \Magento\Framework\Exception\InputException If token or customer id is invalid
      * @throws \Magento\Framework\Exception\NoSuchEntityException If customer doesn't exist
      * @throws \Magento\Framework\Exception\LocalizedException
+     * @deprecated
+     * @see \Magento\Customer\Api\AccountManagementInterface::validateResetPasswordLinkByToken()
      */
     public function validateResetPasswordLinkToken($customerId, $resetPasswordLinkToken);
 
+    /**
+     * Check if password reset token is valid.
+     *
+     * @param string $resetPasswordLinkToken
+     *
+     * @return bool True if the token is valid
+     * @throws \Magento\Framework\Exception\State\InputMismatchException If token is mismatched
+     * @throws \Magento\Framework\Exception\State\ExpiredException If token is expired
+     * @throws \Magento\Framework\Exception\InputException If token is invalid
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function validateResetPasswordLinkByToken($resetPasswordLinkToken);
```

Comment:
Unfortunately, we are not allowed to add new public methods to API interfaces according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
@sidolov I already discussed it with @joni-jones.
We should discuss it with our architects.

Reply:
@VladimirZaets such changes are backward-incompatible anyway. The new method should be defined in scope of the new interface instead of existing.

Reply:
@sidolov  PR was updated. The new method was added into a new interface.